solr/solr-ref-guide/modules/query-guide/pages/highlighting.adoc

```diff
@@ -116,6 +116,17 @@ If set to `true`, only query terms aligning with the field being highlighted wil
 If the query references fields different from the field being highlighted and they have different text analysis, the query may not highlight query terms it should have and vice versa.
 The analysis used is that of the field being highlighted (`hl.fl`), not the query fields.
 
+`hl.queryFieldPattern`::
++
+[%autowidth,frame=none]
+|===
+|Optional |Default: `none`
+|===
++
+Similar to `hl.requireFieldMatch` but allows for multiple fields to match e.g. `q=fieldA:one OR fieldB:two OR fieldC:three` `hl.fl=fieldA` `hl.queryFieldPattern=fieldA,fieldB`
```

Comment:
Does this supersede hl.requireFieldMatch when set?

Reply:
No, if both `hl.requireFieldMatch` and `hl.queryFieldPattern` are specified then only the former one is used and the latter is (silently) ignored.

Reply:
docs updated in https://github.com/apache/solr/pull/757/commits/ee6f1e3e15457d8e89deaf5e68abba199568e586 and also in https://github.com/apache/solr/pull/757/commits/44bd2201f834dd7ef35773e1e3630cf5838ca2b2 added test for it.